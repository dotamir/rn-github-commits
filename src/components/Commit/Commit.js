import React from 'react';
import {StyleSheet} from 'react-native';
import {ListItem, Left, Right, Body, Text, Thumbnail} from 'native-base';

const Commit = ({avatar, message, date, username}) => {
	return (
		<ListItem style={styles.listItem} avatar>
			<Left>
				<Thumbnail
					style={styles.avatar}
					source={{
						height: 50,
						width: 50,
						uri: avatar,
					}}
				/>
			</Left>
			<Body>
				<Text>{username}</Text>
				<Text note>{message}</Text>
			</Body>
			<Right>
				<Text note>{date}</Text>
			</Right>
		</ListItem>
	);
};

const styles = StyleSheet.create({
	avatar: {
		width: 40,
		height: 40,
	},
});

export default Commit;
