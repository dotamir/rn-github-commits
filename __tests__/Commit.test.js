/**
 * @format
 */

import 'react-native';
import React from 'react';
import Commit from './../src/components/Commit/Commit';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
	const commit = {
		avatar: 'https://avatars2.githubusercontent.com/u/16477296?s=460&v=4',
		message: 'Commit message',
		date: '20/11 - 03:43',
		username: 'dotamir',
	};
	const component = renderer.create(<Commit {...commit} />).toJSON();
	expect(component).toMatchSnapshot();
});
