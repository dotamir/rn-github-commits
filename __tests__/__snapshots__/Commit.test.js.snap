// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    avatar={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 16,
        "paddingBottom": null,
        "paddingRight": null,
        "paddingTop": null,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "flex-start",
          "flex": 0,
          "flexDirection": "row",
          "paddingTop": 14,
        }
      }
    >
      <Image
        source={
          Object {
            "height": 50,
            "uri": "https://avatars2.githubusercontent.com/u/16477296?s=460&v=4",
            "width": 50,
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 28,
              "height": 56,
              "width": 56,
            },
            Object {
              "height": 40,
              "width": 40,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": null,
          "alignSelf": null,
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 1,
          "marginLeft": 15,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        uppercase={false}
      >
        dotamir
      </Text>
      <Text
        note={true}
        style={
          Object {
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "200",
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        uppercase={false}
      >
        Commit message
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "stretch",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 0,
          "paddingBottom": 10,
          "paddingLeft": null,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        note={true}
        style={
          Object {
            "alignSelf": null,
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "200",
          }
        }
        uppercase={false}
      >
        20/11 - 03:43
      </Text>
    </View>
  </View>
</View>
`;
